subconscious/model.py

```diff
@@ -324,7 +355,25 @@ def get_index_key(cls, column_name):
             User.get_by(db, age=23, name="guido")
 
         """
-        for key in await cls._get_ids_filter_by(db, **kwargs):
+        offset = kwargs.pop('offset', None)
+        limit = kwargs.pop('limit', None)
+
+        if limit and type(limit) is not int:
+            raise InvalidQuery('If limit is supplied it must be an int')
+        if offset and type(offset) is not int:
+            raise InvalidQuery('If offset is supplied it must be an int')
+
+        ids_to_iterate = await cls._get_ids_filter_by(db, **kwargs)
+        if offset:
+            # Using offset without order_by is pretty strange, but allowed
+            if limit:
+                ids_to_iterate = ids_to_iterate[offset:offset+limit]
+            else:
+                ids_to_iterate = ids_to_iterate[offset:]
+        elif limit:
+            ids_to_iterate = ids_to_iterate[:limit]
```

Comment:
Since this is a copy-paste of the `.all()` limit/offset code, I'd prefer a way to keep this DRY. Perhaps using your new `.query()` syntax it would be possible to only define this once?